### ISSUE TITLE
Exporting Blocks Should Check For missing dependencies

### ISSUE THREAD

**Issue by cycomachead** [jmoenig/Snap--Build-Your-Own-Blocks#882]
I know this is a harder one to solve but it would be nice if Snap<i>!</i> could resolve the dependency tree of custom blocks when exporting. I'm assuming, this would require a recursive search over the selected custom blocks to find which are missing, so this could (potentially) be a bit slow if there were too many blocks, but it would be handy.

**Comment by jmoenig** [comment]
Right. Basically we want to be able to check every custom block for its callers (other custom block definition) and "users" (sprite scripts that contain instances of its definition). It's not hard to do at all, but don't go ahead implementing it just yet, because the whole block-definition-instance relationship mechanism might fundamentally change for macros and OOP...
